doc/extension.rst

```diff
@@ -4,9 +4,9 @@ C Autodoc Extension
 ===================
 
 Hawkmoth provides a Sphinx extension that adds :ref:`new directives
-<directives>` to the Sphinx :any:`C domain <sphinx:c-domain>` to incorporate
-formatted C source code comments into a document. Hawkmoth is Sphinx
-:any:`sphinx:sphinx.ext.autodoc` for C.
+<directives>` to the Sphinx :any:`C domain <c-domain>` to incorporate formatted
```

Comment:
I'd have preferred if these were a separate commit, if not in fact all '3' items in the message, but it's simple enough not to matter.

However, I don't see a reason to remove the explicit source of the tag. Looking at the docs I see no counter indication and explicit > implicit any time of day for me.

There's an `:external:` role as well (don't remember seeing that before), but that one is redundant and less specific than scoping these on each external link.

Reply:
Note that I don't know which part of the docs mentioned in the commit ([2]) is relevant here and I'm commenting under the assumption that it just isn't.

Locally the old references were good for me. Maybe a Sphinx v6 thing?

Reply:
> I'd have preferred if these were a separate commit, if not in fact all '3' items in the message, but it's simple enough not to matter.

Yeah, I don't disagree. Usually it's a bad sign if the commit message contains lists of things (or a sentence beginning with "Also, ").

> However, I don't see a reason to remove the explicit source of the tag. Looking at the docs I see no counter indication and explicit > implicit any time of day for me.
> 
> There's an `:external:` role as well (don't remember seeing that before), but that one is redundant and less specific than scoping these on each external link.

The thing is, I couldn't find anything to back up our current usage of the explicit external references:

```
:any:`sphinx:foo`
```

Per the interpshinx docs the above should be:

```
:external+sphinx:any:`foo`
```

(The `any` role isn't significant here.)

I'm with you on being explicit rather than implicit, but that's just gets pretty verbose. I'm now leaning towards using it anyway, but I wonder where the current usage comes from, and why it works.

Reply:
I'll respin this, with explicit intersphinx references, but I don't know which style to go for.

Reply:
> The thing is, I couldn't find anything to back up our current usage of the explicit external references:
> 
> ```
> :any:`sphinx:foo`
> ```
> 
> Per the interpshinx docs the above should be:
> 
> ```
> :external+sphinx:any:`foo`
> ```
> 
> (The `any` role isn't significant here.)
> 
> I'm with you on being explicit rather than implicit, but that's just gets pretty verbose. I'm now leaning towards using it anyway, but I wonder where the current usage comes from, and why it works.

Good points... I was also surprised I didn't see that syntax in the docs you linked, but didn't go hunting down why it was so. Maybe from an older version which is going out of style.

I'd still make the point that `external+sphinx` is better if we can't verify the old syntax, even if very verbose.